src/daemon/ubuntu_image_host.cpp

```diff
@@ -220,28 +224,39 @@ std::vector<std::string> mp::UbuntuVMImageHost::supported_remotes()
 {
     std::vector<std::string> supported_remotes;
 
-    for (const auto& remote : remotes)
+    for (const auto& [remote_name, _] : remotes)
     {
-        supported_remotes.push_back(remote.first);
+        supported_remotes.push_back(remote_name);
     }
 
     return supported_remotes;
 }
 
 void mp::UbuntuVMImageHost::fetch_manifests()
 {
-    for (const auto& remote : remotes)
+    for (const auto& [remote_name, remote_info] : remotes)
     {
         try
         {
-            check_remote_is_supported(remote.first);
+            check_remote_is_supported(remote_name);
+            auto official_site = remote_info.get_official_url();
+            auto manifest_bytes_from_official = download_manifest(official_site, url_downloader);
```

Comment:
Wouldn't it be better to *only* download the official manifest if the mirror url config setting is not set?  This would save a little bandwidth and time fetching a manifest that won't be used.

Reply:
If the mirror url config setting is not set, the manifest would be `std::nullopt`. It does download manifest from official  host only.
https://github.com/rim99/multipass/blob/57711ba301ba175265bfe35bf499501bcd623658/src/daemon/ubuntu_image_host.cpp#L246

Reply:
Ok, I've read your reasoning and I understand your points.  So really, this is *preferring* a mirror if it exists and is a copy of the official site.  One thing I'm a bit concerned about with this approach is if a user is behind a firewall that prevents access to the official site, but allows access to the mirror site.  This wouldn't work for this approach.  But I guess this is no worse than what is already there in that the official site can't be reached anyways and the user can't use Multipass.

Thank you for the explanation and I'll give it some more thought.